src/HackF5.UnitySpy/AssemblyImageFactory.cs

```diff
@@ -103,9 +103,9 @@ private static AssemblyImage GetAssemblyImage(UnityProcessFacade process, string
                 domain = process.ReadPtr(domainAddress);
             }
             //// pointer to array of structs of type _MonoAssembly
-            var assemblyArrayAddress = process.ReadPtr(domain + process.MonoLibraryOffsets.ReferencedAssemblies);
+            var assemblyArrayAddress = process.ReadPtr(domain + /*process.MonoLibraryOffsets.ReferencedAssemblies*/160);
             for (var assemblyAddress = assemblyArrayAddress;
-                assemblyAddress != Constants.NullPtr;
+                assemblyAddress != IntPtr.Zero;
```

Comment:
Just curious, why was this necessary?

Reply:
To clarify: I'm all ok for simplification (in this case it indeed doesn't really make a lot of sense to have a Constants class just for the 0 pointer), but I would rather that this go in a separate PR, so that this PR only contains the code that is needed for the new version.

Reply:
It's part of the modifications I had to make it easier for me to work on the PR. I removed most of them when creating the PR but since `Constants.NullPtr` was just used as an alias for `IntPtr.Zero` might as well let that change here 🤷 I can remove it if needed

Reply:
If possible I'd like to keep this PR for the modifications that are strictly necessary.
You can open another PR with this change only, and I can merge it first, so you won't have to revert the changes in this PR

Reply:
Created #21